src/Symfony/Component/HttpKernel/Controller/ControllerResolver.php

```diff
@@ -112,21 +120,43 @@ protected function doGetArguments(Request $request, $controller, array $paramete
         foreach ($parameters as $param) {
             if (array_key_exists($param->name, $attributes)) {
                 $arguments[] = $attributes[$param->name];
-            } elseif ($param->getClass() && $param->getClass()->isInstance($request)) {
-                $arguments[] = $request;
-            } elseif ($param->isDefaultValueAvailable()) {
-                $arguments[] = $param->getDefaultValue();
-            } else {
-                if (is_array($controller)) {
-                    $repr = sprintf('%s::%s()', get_class($controller[0]), $controller[1]);
-                } elseif (is_object($controller)) {
-                    $repr = get_class($controller);
-                } else {
-                    $repr = $controller;
+
+                continue;
+            }
+
+            if ($class = $param->getClass()) {
+                if ($class->isInstance($request)) {
+                    $arguments[] = $request;
+
+                    continue;
                 }
 
-                throw new \RuntimeException(sprintf('Controller "%s" requires that you provide a value for the "$%s" argument (because there is no default value or because there is a non optional argument after this one).', $repr, $param->name));
+                if ($class->implementsInterface('Psr\Http\Message\ServerRequestInterface')) {
+                    if (null === $this->httpMessageFactory) {
+                        throw new \RuntimeException('The PSR-7 Bridge must be installed to inject instances of Psr\Http\Message\ServerRequestInterface in controllers.');
+                    }
+
+                    $arguments[] = $this->httpMessageFactory->createRequest($request);
```

Comment:
I don't understand how this works. The Symfony Request won't be in the RequestStack.